Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/GameUI.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections;
+using TMPro;
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the UI of the "NetworkVariable vs RPCs" scene
+    /// </summary>
+    public class GameUI : MonoBehaviour
+    {
+        [SerializeField] TMP_Text startupLabel;
+        [SerializeField] TMP_Text controlsLabel;
+
+        void OnEnable()
+        {
+            Refreshlabels(NetworkManager.Singleton && (NetworkManager.Singleton.IsClient || NetworkManager.Singleton.IsServer));
+            StartCoroutine(SubscribeToNetworkManagerEvents());
+        }
+
+        IEnumerator SubscribeToNetworkManagerEvents()
+        {
+            yield return new WaitUntil(() => NetworkManager.Singleton);
+            NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnectedCallback;
+            NetworkManager.Singleton.OnServerStarted += OnServerStarted;
+            NetworkManager.Singleton.OnClientDisconnectCallback += OnClientDisconnectCallback;
+        }
+
+        void OnDestroy()
+        {
+            if (NetworkManager.Singleton)
+            {
+                NetworkManager.Singleton.OnClientConnectedCallback -= OnClientConnectedCallback;
+                NetworkManager.Singleton.OnServerStarted -= OnServerStarted;
+                NetworkManager.Singleton.OnClientDisconnectCallback -= OnClientDisconnectCallback;
+            }
+        }
+
+        void OnServerStarted()
+        {
+            OnNetworkedSessionStarted();
+        }
+
+        void OnClientConnectedCallback(ulong obj)
+        {
+            if (NetworkManager.Singleton && NetworkManager.Singleton.IsServer)
+            {
+                return; //we don't want to do actions twice when playing as a host
+            }
+            OnNetworkedSessionStarted();
```

Comment:
same here, no need for the wrapper when calling a method with an argument is the same. You could just call RefreshLabels(true). it'd make things simpler and easier to follow.
If you're trying to be future proof with having special started/ended events, in our case here I'd say yagni, there's little chance those samples will grow. Plus OnClientConnectedCallback sort of already tells you this.

Reply:
1. `OnNetworkedSessionStarted` is there mainly for clarity. Its implementation is not really relevant, what's relevant is showing how to call a method when a session is started, regardless of where you are calling it from
2. We don't know how OnNetworkedSessionStarted will grow with future dioramas, and it already saves us from managing an extra parameter

Reply:
dunno what I can say more than I already did.
1. OnClientConnectedCallback sort of already tells you this.
2. pre-optimization is the root of all evil. We can refactor when we get there (if this particular ProximityCheck sample gets more code added to it)

Reply:
1. `OnClientConnectedCallback` tells you the client connected. It doesn't cover the case when the server connected too.
2. We're not pre-optimizing here, we're avoiding code duplication. Instead of calling a method with a parameter twice, we're calling a method with no parameters and increased semantic value.

It's ok if you disagree due to personal opinions such as "_it'd make things simpler and easier to follow._", we can't agree on everything. 
Should I consider this a blocker, or can I resolve the conversation?

Reply:
I wouldn't call this a personal opinion, I would call this a word of advice on good practices. DRY doesn't suggest not repeating code, but not repeating knowledge. You can check out the inventor of the DRY principle clarifying this here https://youtu.be/taCNjiiusRk?t=1365 it's a common misconception worth flagging as a general practice

> The code is the same, but the knowledge they represent is different. The two functions validate two separate things that just happen to have the same rules. That's a coincidence, not a duplication.

Reply:
Since this was considered a blocker, I addressed it at: 93667685